app/controllers/application_controller.rb

```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  helper_method :current_user
+  
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```

Comment:
1. What does `||=` operator do?
2. What's the difference between `find` and `find_by`